### PR TITLE
Bugzilla credentials moved to config

### LINKED ISSUE #4045: Add Buzilla credentials on configuration
After playing some time with the Bugzilla API, I discovered "flags" is only available on authenticated calls. I opened a bug for it: https://bugzilla.redhat.com/show_bug.cgi?id=1399755. So I am going to add bugzilla credentials on robottelo.properties accordingly.

related to #3868

### PR DESCRIPTION
**Pull request by renzon**
closes #4045

Before merging this, I have to add BZ configuration on jenkins.

[Edit]

Configuration added: https://github.com/SatelliteQE/robottelo-ci/pull/421. PR good to go.

### CODE REVIEW

**Comment by renzon** [comment]
I was checking and noted our travis was not running any test. The comment point foreman test which need a sat instance which I agree with. But robottelo (unit tests of framework itself). Run very quick and do not depend on external satellite.

**Comment by renzon** [comment]
robottello tests results:

```console
(robottelo) [renzo@note robottelo]$ py.test tests/robottelo/
========================================================== test session starts ==========================================================
platform linux2 -- Python 2.7.12, pytest-3.0.3, py-1.4.31, pluggy-0.4.0
rootdir: /home/renzo/PycharmProjects/robottelo, inifile: 
plugins: xdist-1.15.0, cov-2.3.1
collected 203 items 

tests/robottelo/test_api_utils.py ..
tests/robottelo/test_cli.py ......................................
tests/robottelo/test_config_casts.py ..............
tests/robottelo/test_config_settings.py .....
tests/robottelo/test_datafactory.py .....
tests/robottelo/test_decorators.py ...........................................
tests/robottelo/test_hammer.py .........
tests/robottelo/test_helpers.py .............
tests/robottelo/test_host_info.py ................
tests/robottelo/test_ssh.py ..............
tests/robottelo/test_ui.py .....
tests/robottelo/test_ui_locators.py ........................
tests/robottelo/test_vm.py ......
tests/robottelo/decorators/test_host.py .........

====================================================== 203 passed in 1.43 seconds ==========================================
```

**Comment by renzon** [comment]
On test using skip_bz decorator still working:

```console
pyt.test -k OperatingSystemTestCase and test_negative_create_with_too_long_description"

============================= 34 tests deselected ==============================
=================== 1 passed, 34 deselected in 12.90 seconds ===================
```

File: robottelo/config/base.py [hunk 1]
```diff
@@ -740,56 +786,19 @@ def configure(self):
...
+            dir(self)
+        )
+        feature_settings = filter(
+            lambda tpl: isinstance(tpl[1], FeatureSettings),
+            attrs
+        )
+        for name, settings in feature_settings:
+            if self.reader.has_section(name) or name == 'server':
+                settings.read(self.reader)
+                self._validation_errors.extend(settings.validate())
```
Comment (rochacbruno) [inline robottelo/config/base.py]: :+1:

File: robottelo.properties.sample [hunk 2]
```diff
@@ -104,6 +103,16 @@ ssh_key=
 # capsule_repo=http://capsule/repo
 # sattools_repo=http://sattools/repo
 
+# Bugzilla data bellow are used to skip tests which are not implemented
```
Comment (sthirugn) [inline robottelo.properties.sample]: s/bellow/below

File: robottelo.properties.sample [hunk 3]
```diff
@@ -104,6 +103,16 @@ ssh_key=
...
+[bugzilla]
+# buzilla URL
```
Comment (sthirugn) [inline robottelo.properties.sample]: s/buzilla/bugzilla

**Comment by rochacbruno** [comment]
ACK
(remember that this needs to be cherry picked)

File: robottelo/decorators/__init__.py [hunk 4]
```diff
@@ -267,18 +266,26 @@ def _get_bugzilla_bug(bug_id):
     if bug_id in _bugzilla:
         LOGGER.debug('Bugzilla bug {0} found in cache.'.format(bug_id))
     else:
+        if settings.bugzilla.url is None:
+            _bugzilla[bug_id] = None
+            LOGGER.warning('Bugzilla url not defined on robotello.properties')
```
Comment (sthirugn) [inline robottelo/decorators/__init__.py]: s/robotello/robottelo

File: robottelo.properties.sample [hunk 5]
```diff
@@ -104,6 +103,16 @@ ssh_key=
...
+# bz_url=https://bugzilla.redhat.com
```
Comment (sthirugn) [inline robottelo.properties.sample]: bugzilla url is going to be a constant - there is no need to have this in the robottelo.properties file.  I vote to leaving this in robottelo/decorators/__init__.py : BUGZILLA_URL = "https://bugzilla.redhat.com/xmlrpc.cgi"

Comment (sthirugn) [inline robottelo.properties.sample]: if you want to move that constant to robottelo/config/base.py - it is okay too

**Comment by renzon** [comment]
@sthirugn I have made all typo corrections you pointed out. About url been constant, I thought about it and once ids are related to RH BZ, you are right. But I never tested it locally like @rochacbruno  mentioned. @rochacbruno that is a common use case?

**Comment by sthirugn** [comment]
Personally, I go with a simple rule - if a value needs to be changed by the end user - specify that in a config file (robottelo.properties in this case).  If something you know is a constant for life, why expose that in the user config? Note: We also have a constants.py for these kind of values.

**Comment by renzon** [comment]
@sthirugn In fact I thought that if one fork robotello to test its own for with its own bz, one should only change data which is in properties. But that is so far from a real usecase that I must agree with you ;). I am going to move BZ back to module constant and raise PR to jenkins properties get credentials as well.

**Comment by renzon** [comment]
@sthirugn In fact I thought that if one fork robotello to test its own for with its own bz, one should only change data which is in properties. But that is so far from a real usecase that I must agree with you ;). I am going to move BZ back to module constant and raise PR to jenkins properties get credentials as well.

**Pull request merged**